### DISCUSSION TITLE
All examples failing with mock

### DISCUSSION THREAD

**Discussion by biancaganescu** [zkonduit/ezkl#224] [Q&A] [answered]
Hi, on my end all examples throw "error: constraint not satisfied" for both the latest commit and the commit on 14th of April that I was initially using (a3017cf) when using the mock command. For example, running 1l_div throws:
```
[0s, ezkl] - command: 
 |   {
 |    "args": {
 |      "allocated_constraints": null,
 |      "bits": 16,
 |      "check_mode": "SAFE",
 |      "logrows": 17,
 |      "pack_base": 1,
 |      "public_inputs": false,
 |      "public_outputs": true,
 |      "public_params": false,
 |      "scale": 7,
 |      "tolerance": 0
 |    },
 |    "command": {
 |      "Mock": {
 |        "data": "input.json",
 |        "model": "network.onnx"
 |      }
 |    }
 |  }
[*] [0s, ezkl_lib::commands] - loading data from input.json
[*] [0s, ezkl_lib::graph::model] - dummy model layout
[*] [0s, ezkl_lib::graph::model] - model outputs are nodes: [2]
[*] [0s, ezkl_lib::graph] - public inputs lengths: [1]
[*] [0s, ezkl_lib::execute] - Mock proof
[*] [0s, ezkl_lib::graph::model] - configuring model
[*] [0s, ezkl_lib::graph::model] - model layout
[*] [0s, ezkl_lib::graph::model] - model outputs are nodes: [2]
[*] [0s, ezkl_lib::graph::model] - model outputs are nodes: [2]
[*] [0s, ezkl_lib::graph::model] - computing...
error: constraint not satisfied

  Cell layout in region 'model':
    | Offset | A1 | A2 |
    +--------+----+----+
    |    2   | x0 | x1 | <--{ Gate 'RANGE' applied here

  Constraint '':
    (S5 * (0x2 - S5)) * (x0 - x1) = 0

  Assigned cell values:
    x0 = 0x68
    x1 = 0x62

thread 'main' panicked at 'circuit was not satisfied', /home/bianca/.cargo/git/checkouts/halo2-afe2d0b0be6b3c3a/0d56c57/halo2_proofs/src/dev.rs:1453:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

When the "output_data" in the input.json is empty the run succeeds for the commit on April the 14th (a3017cf).>

Tested on both a mac and linux machine, reinstalled everything from scratch.

**Comment by alexander-camuto** [comment]
Heya @biancaganescu can you give me some more details ? we run all of these tests in our CI pipeline and they are passing. Also ran them locally and they are fine. Can you let me know if all the tests run by `cargo test tests::mock_` pass on your local machine and remote linux ? 

See here for instance: https://github.com/zkonduit/ezkl/actions/runs/4892091504/jobs/8733936871

**Reply by alexander-camuto** [reply]
You can also message me on twitter if you need a quicker Q - A cycle (twitter is in my github bio)

**Reply by biancaganescu** [reply]
Sure, thank you, I just followed you on twitter to discuss, but your message box does not seem open, I can't message you

**Reply by alexander-camuto** [reply]
Msged u

**Reply by alexander-camuto** [reply, marked as answer]
Issue was locally modified .json files. @lancenonce we need to make it clearer in the docs that the forward pass is required to account for quantization error. 

Bar that pyezkl should account for it when generating input.json.
